coral-presto/src/main/java/com/linkedin/coral/presto/rel2presto/RelToPrestoConverter.java

```diff
@@ -168,6 +170,29 @@ public Result visit(Uncollect e) {
     return result(asNode, ImmutableList.of(Clause.FROM), e, null);
   }
 
+  /**
+   * This overridden function makes sure that the basetable names in the output SQL
+   * will be in the form of "dbname.tablename" instead of "catalogname.dbname.tablename"
+   *
+   * Presto can have configurable catalog names. In that case the HiveToRelConverter's default "hive" catalog will
+   * cause failures:  https://github.com/prestosql/presto/issues/5785. If catalogname is not prepended, presto uses
```

Comment:
curious, where does the prepending happen?

Reply:
Prepending is the default behavior of `RelToSqlConverter::visit(TableScan e)`. We override the behavior here.